org.civicrm.reqangular/src/common/angular-date/src/services/DateFactory.js

```diff
@@ -20,13 +23,33 @@ function DateFactory() {
                 ];
             }
 
-            if(typeof strict === 'undefined'){
+            if (typeof strict === 'undefined') {
                 strict = true;
             }
 
             return moment(dateString, format, strict);
+        },
+
+        /**
+         * @description Returns Current Date Format
+         * @returns {string}
+         */
+        getDateFormat: function () {
+            return this.dateFormat;
+        },
+
+        /**
+         * @description Fetches date format setting from
+         * @returns {Promise}
+         */
+        fetchDateFormatFromSettings: function(){
```

Comment:
> Also, I'd need to return string or a promise object depending on this condition 

That's not necessary: the method will still be async no matter if it actually needs to fetch data or not, because the data will eventually be wrapped in a Promise, which would make the method's behaviour and consumption consistent (either it waits for the data from the server to arrive, or just the next tick to immediately return the data already available).

This way the consumer of the method would not have to worry about its actual implementation, she would just need to know that it returns a Promise:
```js
DateFactory.getDateFormat().then(function (format) {
    format; // DD/MM/YYYY
});
```
The method's implementation would be something like:
```js
getDateFormat: function () {
    if (this.dateFormat) {
        // wraps the data in a Promise and resolves it
        return $q.when(this.dateFormat);
    } else {
        // makes an actual async request
        return $http({ /* fetchData */ }).then(function (format) {
            this.dateFormat = format;
            return this.dateFormat;
        });
    }
}
```